File: src/main/java/committee/nova/migration/bswk/BasaltWalker.java
package committee.nova.migration.bswk;

import committee.nova.migration.bswk.common.block.SoftBasaltBlock;
import committee.nova.migration.bswk.common.enchantment.BasaltWalkerEnchantment;
import net.fabricmc.api.ModInitializer;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

public class BasaltWalker implements ModInitializer {
    public static final String MODID = "bswk";
    public static BasaltWalkerEnchantment enchantment;
    public static SoftBasaltBlock block;

    public static Identifier prefixId(String id) {
        return new Identifier(MODID, id);
    }

    @Override
    public void onInitialize() {
        block = Registry.register(Registry.BLOCK, prefixId("soft_basalt"), new SoftBasaltBlock());
        enchantment = Registry.register(Registry.ENCHANTMENT, prefixId("basalt_walker"), new BasaltWalkerEnchantment());
    }
}

File: src/main/scala/committee/nova/migration/bswk/BasaltWalker.scala
package committee.nova.migration.bswk

import committee.nova.migration.bswk.BasaltWalker.proxy
import committee.nova.migration.bswk.common.proxy.CommonProxy
import net.minecraftforge.fml.common.Mod.EventHandler
import net.minecraftforge.fml.common.event.{FMLInitializationEvent, FMLPreInitializationEvent}
import net.minecraftforge.fml.common.{Mod, SidedProxy}

object BasaltWalker {
  final val MODID = "bswk"
  @SidedProxy(serverSide = "committee.nova.migration.bswk.common.proxy.CommonProxy")
  val proxy: CommonProxy = new CommonProxy
  @Mod.Instance(BasaltWalker.MODID)
  var instance: BasaltWalker = _
}

@Mod(modid = BasaltWalker.MODID, useMetadata = true)
class BasaltWalker {
  BasaltWalker.instance = this

  @EventHandler
  def preInit(event: FMLPreInitializationEvent): Unit = proxy.preInit(event)

  @EventHandler
  def init(event: FMLInitializationEvent): Unit = proxy.init(event)
}